test/peer-id.spec.js

```diff
@@ -86,6 +86,15 @@ describe('PeerId', () => {
     expect(id.marshalPubKey()).to.deep.equal(id2.marshalPubKey())
   })
 
+  it('Recreate from Protobuf', async () => {
+    const id = await PeerId.createFromProtobuf(testId.marshaled)
+    expect(testIdB58String).to.equal(id.toB58String())
+    const encoded = Buffer.from(testId.privKey, 'base64')
+    const id2 = await PeerId.createFromPrivKey(encoded)
+    expect(testIdB58String).to.equal(id2.toB58String())
+    expect(id.marshalPubKey()).to.deep.equal(id2.marshalPubKey())
+  })
```

Comment:
@mkg20001 it looks like `marshal` isn't being tested, can we validate that marshaling the peerid gets us the same `testId.marshaled` value?

Other than that, this looks good.

Reply:
Done